script/hassfest/bluetooth.py

```diff
@@ -0,0 +1,73 @@
+"""Generate bluetooth file."""
+from __future__ import annotations
+
+import pprint
+import re
+
+from .model import Config, Integration
+
+BASE = """
+\"\"\"Automatically generated by hassfest.
+
+To update, run python3 -m script.hassfest
+\"\"\"
+from __future__ import annotations
+
+# fmt: off
+
+BLUETOOTH: list[dict[str, str | int]] = {}
+""".strip()
+
+
+def generate_and_validate(integrations: list[dict[str, str]]):
+    """Validate and generate bluetooth data."""
+    match_list = []
+
+    for domain in sorted(integrations):
+        integration = integrations[domain]
+
+        if not integration.manifest or not integration.config_flow:
+            continue
+
+        match_types = integration.manifest.get("bluetooth", [])
+
+        if not match_types:
+            continue
+
+        for entry in match_types:
+            match_list.append({"domain": domain, **entry})
+
+    # JSON will format `True` as `true`
+    # re.sub for flake8 E128
+    formatted = pprint.pformat(match_list)
+    formatted_aligned_continuation = re.sub(r"^\[\{", "[\n {", formatted)
+    formatted_align_indent = re.sub(
```

Comment:
You could also import Black :)

```python
import black

formatted = black.format_str(formatted, mode=black.Mode())
```

Reply:
tried that, ends up in a loop

```
black....................................................................Failed
- hook id: black
- files were modified by this hook
```

Reply:
```diff
diff --git a/homeassistant/generated/bluetooth.py b/homeassistant/generated/bluetooth.py
index 1d6340de98..038dd06304 100644
--- a/homeassistant/generated/bluetooth.py
+++ b/homeassistant/generated/bluetooth.py
@@ -6,6 +6,5 @@ from __future__ import annotations
 
 # fmt: off
 
-BLUETOOTH: list[dict[str, str | int]] = [
-    {'domain': 'switchbot',
-     'service_uuid': 'cba20d00-224d-11e6-9fb8-0002a5d5c51b'}]
+BLUETOOTH: list[dict[str, str | int]] = [{"domain": "switchbot", "service_uuid": "cba20d00-224d-11e6-9fb8-0002a5d5c51b"}]
+

```

Reply:
I coped this from dhcp

Reply:
resolved by refactoring.  We don't need the dhcp true fix